apps/allocations/app/components/Form/Input/OptionsInput.js

```diff
@@ -18,14 +18,17 @@ class OptionsInput extends React.Component {
     name: PropTypes.string.isRequired,
     value: PropTypes.array.isRequired,
     onChange: PropTypes.func.isRequired,
+    validator: PropTypes.func,
+    error: PropTypes.bool
   }
 
   addOption = () => {
     // TODO: Implement some rules about what an 'Option can be' duplicates, etc
     const { input, name, value } = this.props
-    if (input && !value.includes(input)) {
+    if (input && !value.includes(input) && this.props.validator(input)) {
       this.props.onChange({ target: { name, value: [...value, input] } })
       this.props.onChange({ target: { name: 'optionsInput', value: '' } })
+      this.props.error = true
```

Comment:
You generally don't want to do this; you probably want to have a component set its state instead (changing the prop like this doesn't cause a rerender, and the next rerender might overwrite these existing props leading to this state being lost).

Reply:
OptionsInput was refactored to SFC and props were reviewed